src/main/java/com/michalsydoryk/app/boardchecker/BoardCheckerAbstract.java

```diff
@@ -0,0 +1,15 @@
+package com.michalsydoryk.app.boardchecker;
+
+import java.util.LinkedList;
+import java.util.List;
+
+abstract class BoardCheckerAbstract<T, Q>  implements BoardChecker<Q>{
```

Comment:
Ok now I understand, interface -> abstract class -> implementation. Reconsider using only abstract class with abstract check method from interface. If you don't want to do this prove me why this solution is better:)